### PR TITLE
Runnable samples

### PR DESCRIPTION
**Pull request by scothis**
There are four samples, each in their own directory:
- Spring PetClinic with MySQL (./samples/spring-petclinic)
- Controlled Resource (./samples/controlled-resource)
- Overridden Type and Provider (./samples/overridden-type-provider)
- Multiple Bindings (./samples/multi-binding)

Signed-off-by: Scott Andrews <andrewssc@vmware.com>

### CODE REVIEW

**Comment by scothis** [comment]
Originally from https://github.com/vmware-tanzu/servicebinding/tree/main/samples

File: samples/spring-petclinic/workload.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      app: spring-petclinic
+  template:
+    metadata:
+      labels:
+        app: spring-petclinic
+    spec:
+      containers:
+        - name: workload
+          # built with paketo builder from https://github.com/spring-projects/spring-petclinic
+          image: ghcr.io/vmware-tanzu/servicebinding/test/petclinic:20200922
```
Comment (baijum) [inline samples/spring-petclinic/workload.yaml:47]: Should we push this image to `ghcr.io/servicebinding`? Also looks like the image is from 2020. A new build would be nice.

**Review by baijum — APPROVED** [review]

File: samples/spring-petclinic/workload.yaml (hunk 1 above)
Comment (scothis) [reply samples/spring-petclinic/workload.yaml]: updated the image ref to ghcr.io/servicebinding/runtime/samples/spring-petclinic@sha256:83ab44832a1db6c03d34e758199b0d9cbf29ce5beeaac4fbf96443a63342b3d4

> A new build would be nice.

It's a sample, the age of the image isn't important as long as it works. If anything, showing that an older image still works is a good thing.

Comment (baijum) [reply samples/spring-petclinic/workload.yaml]: ok, thanks!

**Pull request merged**